test/jdk/javax/rmi/ssl/SSLSocketParametersTest.java

```diff
@@ -89,117 +73,67 @@ public void runClient(Remote stub) throws IOException {
         }
     }
 
-    public static class ClientFactory extends SslRMIClientSocketFactory {
-
-        public ClientFactory() {
-            super();
-        }
-
-        public Socket createSocket(String host, int port) throws IOException {
-            System.out.println("ClientFactory::Calling createSocket(" +
-                               host + "," + port + ")");
-            return super.createSocket(host, port);
-        }
+    public void testRmiCommunication(RMIServerSocketFactory serverSocketFactory) throws Exception {
+        HelloImpl server = new HelloImpl();
+        Hello stub = (Hello)UnicastRemoteObject.exportObject(server,
+                0, new SslRMIClientSocketFactory(), serverSocketFactory);
+        HelloClient client = new HelloClient();
+        client.runClient(stub);
```

Comment:
You could even inline HelloClient here, as I don't think that having a separate class is actually doing anything..

Reply:
it does emphasize the classic RMI client/server scenario!